specification/hugr.md

```diff
@@ -569,19 +640,70 @@ analysis required to move computations out of a CFG-node into
 Conditional- and TailLoop-nodes). Note that such conversion could be
 done for only a subpart of the HUGR at a time.
 
-**Example CFG (TODO update with** `Sum` **types)** the following CFG is
-equivalent to the previous example. Besides the use of Ext
-edges to reduce passing of P and X, I have also used the normalization
-of moving operations out of the exit-block into the surrounding graph;
-this results in the qubit being passed right through so can also be
-elided. Further normalization of moving F out of the entry-block into
-the surrounding graph is also possible. Indeed every time a SESE region
+The following CFG is equivalent to the previous example. In this diagram:
+
+* the dotted arrow from "angle source" to "F" is an `Ext` edge (from an
+  ancestral DFG into the CFG's entry block);
+* the dotted arrow from "F" to "G" is a `Dom` edge (from a dominating basic
+  block);
+* the `Rz` operation has been moved outside the CFG into the surrounding DFG, so
+  the qubit does not need to be passed in to the CFG.
+
+As a further normalization it would be possible to move F out of the CFG.
+Alternatively, as an optimization it could be moved into the BB1 block.
+
+Indeed every time a SESE region
 is found within a CFG (where block *a* dominates *b*, *b* postdominates
 *a*, and every loop containing either *a* or *b* contains both), it can
 be normalized by moving the region bracketted by *a…b* into its own
 CFG-node.
 
-<img src="attachments/2647818241/2647818458.png" width="512px">
+```mermaid
+flowchart
+    subgraph CFG
+        direction TB
+        subgraph Entry
+            direction TB
+            EntryIn["Input"] -- "P" --> Entry_["P?"]
+            Entry_ -- "[()|(P)]" --> EntryOut["Output"]
+            F
+        end
+        subgraph BB1
+            direction TB
+            BB1In["Input"] -- "(Order)" --> BB1_["..."]
+            BB1_ -- "[()]" --> BB1Out["Output"]
+            G -- "angle" --> BB1Out
+        end
+        subgraph BB2
```

Comment:
So most of the layout problem here looks to stem from mermaid putting BB1 as starting above BB2 (and adding a bunch of space inside BB1 at the top to make it high enough). I don't know why it does that, but I wonder if shuffling the order of BB1/BB2 might help?

Reply:
That makes no difference whatsoever, but let me see if there's anything else I can do to make it look more sensible.

Reply:
No joy, will make a github issue to revisit this.